crates/bevy_ecs/src/query/access.rs

```diff
@@ -387,6 +393,54 @@ impl<T: SparseSetIndex> From<FilteredAccess<T>> for FilteredAccessSet<T> {
     }
 }
 
+/// Records how two accesses conflict with each other
+#[derive(Debug, PartialEq)]
+pub enum AccessConflict {
```

Comment:
nit: maybe `AccessConflicts` since this represents a set of intersections?

maybe even `AccessIntersection` itself...